### DISCUSSION TITLE
Reading queue length from another pool

### DISCUSSION THREAD

**Discussion by cnees** [sidekiq/sidekiq#5269] [Ideas]
I can use
```
Sidekiq::Queue.new(MyJob.sidekiq_options['queue']).size
```
to find the size of a queue before I throw more jobs onto it. But if MyJob uses a different Redis client, it'll check the Redis client configured for `Sidekiq` and return the wrong result.

I sense temporarily reconfiguring Sidekiq to use a different Redis pool is kind of risky. What if another thread enqueues jobs to the second Redis instead of the correct one? What if it crashes or branches and fails to reset to the correct default?

It would be nice to be able to do something like this:
```
Sidekiq::Queue.new(queue_name, redis = AlternateRedis).size
```
or this:
```
Sidekiq::Client.via(AlternateRedis) do
  Sidekiq::Queue.new(queue_name).size # This currently runs without error, but doesn't take the `via` pool into account
end
```

**Comment by mperham** [comment]
Are you using Sidekiq Pro or Enterprise?

**Reply by cnees** [reply]
Yes, I'm on Sidekiq Pro.

**Reply by mperham** [reply]
There's no easy wrapper API to do this today but I can add a simple `Sidekiq.via` helper which works as you show. Can you email me privately (mike at contribsys.com) and I will show you the implementation which I will add to the next Pro release?

**Reply by cnees** [reply]
Thanks, done!
